File: Project/Its Quiz Time/ItsQuizTime - Copy/app/src/main/java/com/mrahmed/itsquiztime/Activity/QuestionActivity.java
package com.mrahmed.itsquiztime.Activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Window;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
import com.bumptech.glide.request.RequestOptions;
import com.mrahmed.itsquiztime.Adapter.QuestionAdapter;
import com.mrahmed.itsquiztime.Domain.QuestionModel;
import com.mrahmed.itsquiztime.R;
import com.mrahmed.itsquiztime.databinding.ActivityQuestionBinding;

import java.util.ArrayList;
import java.util.List;

public class QuestionActivity extends AppCompatActivity implements QuestionAdapter.Score {
    private ActivityQuestionBinding binding;
    private int position = 0;
    private List<QuestionModel> receivedList = new ArrayList<>();
    private int allScore = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityQuestionBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        Window window = this.getWindow();
        window.setStatusBarColor(ContextCompat.getColor(this, R.color.gray));

        receivedList = getIntent().getParcelableArrayListExtra("list");

        // Set up views and click listeners
        binding.backBtn.setOnClickListener(v -> finish());

        binding.progressBar.setProgress(1);
        binding.questionTxt.setText(receivedList.get(position).getQuestion());

        int drawableResourceId = getResources().getIdentifier(
                receivedList.get(position).getPicPath(),
                "drawable",
                getPackageName()
        );

        Glide.with(QuestionActivity.this)
                .load(drawableResourceId)
                .centerCrop()
                .apply(RequestOptions.bitmapTransform(new RoundedCorners(60)))
                .into(binding.pic);

        loadAnswer();

        binding.rightArrow.setOnClickListener(v -> {
            if (binding.progressBar.getProgress() == 10) {
                Intent intent = new Intent(QuestionActivity.this, ScoreActivity.class);
                intent.putExtra("Score", allScore);
                startActivity(intent);
                finish();
                return;
            }
            position++;
            binding.progressBar.setProgress(binding.progressBar.getProgress() + 1);
            binding.questionNumberTxt.setText("Question " + binding.progressBar.getProgress() + "/10");
            binding.questionTxt.setText(receivedList.get(position).getQuestion());

            int drawableResourceIdNext = getResources().getIdentifier(
                    receivedList.get(position).getPicPath(),
                    "drawable", getPackageName()
            );

            Glide.with(QuestionActivity.this)
                    .load(drawableResourceIdNext)
                    .centerCrop()
                    .apply(RequestOptions.bitmapTransform(new RoundedCorners(60)))
                    .into(binding.pic);

            loadAnswer();
        });

        binding.leftArrow.setOnClickListener(v -> {
            if (binding.progressBar.getProgress() == 1) {
                return;
            }
            position--;
            binding.progressBar.setProgress(binding.progressBar.getProgress() - 1);
            binding.questionNumberTxt.setText("Question " + binding.progressBar.getProgress() + "/10");
            binding.questionTxt.setText(receivedList.get(position).getQuestion());

            int drawableResourceIdPrev = getResources().getIdentifier(
                    receivedList.get(position).getPicPath(),
                    "drawable", getPackageName()
            );

            Glide.with(QuestionActivity.this)
                    .load(drawableResourceIdPrev)
                    .centerCrop()
                    .apply(RequestOptions.bitmapTransform(new RoundedCorners(60)))
                    .into(binding.pic);

            loadAnswer();
        });

        // Initialize the adapter
        List<String> answers = new ArrayList<>(); // Populate with the answers
        QuestionAdapter adapter = new QuestionAdapter(receivedList.get(position).getCorrectAnswer(), answers, this);
        // Set the adapter to your RecyclerView here
    }

    private void loadAnswer() {
        List<String> users = new ArrayList<>();
        users.add(receivedList.get(position).getAnswer_1());
        users.add(receivedList.get(position).getAnswer_2());
        users.add(receivedList.get(position).getAnswer_3());
        users.add(receivedList.get(position).getAnswer_4());

        // Create and set the adapter for the RecyclerView
        QuestionAdapter questionAdapter = new QuestionAdapter(
                receivedList.get(position).getCorrectAnswer(), users, this
        );
        // Set the adapter to your RecyclerView here
    }

    @Override
    public void amount(int number, String clickedAnswer) {
        allScore = number;
        receivedList.get(position).setClickedAnswer(clickedAnswer);
    }
}

File: Project/QuizTime/app/src/main/java/com/mrahmed/quiztime/Activity/QuestionActivity.kt
package com.mrahmed.quiztime.Activity

import android.content.Intent
import android.os.Bundle
import android.view.Window
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import com.mrahmed.quiztime.Domain.QuestionModel
import com.mrahmed.quiztime.R
import com.mrahmed.quiztime.databinding.ActivityQuestionBinding

class QuestionActivity : AppCompatActivity() {
    private lateinit var binding: ActivityQuestionBinding
    var position: Int = 0
    var receivedList: MutableList<QuestionModel> = mutableListOf()
    var allScore = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityQuestionBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val window: Window = this@QuestionActivity.window
        window.statusBarColor = ContextCompat.getColor(this@QuestionActivity, R.color.gray)

    receivedList = (intent.getParcelableArrayExtra<QuestionModel>("list")!! as MutableList<QuestionModel>).toMutableList()
        binding.apply {
            backBtn.setOnClickListener{finish()}

            progressBar.progress=1

            questionTxt.text = receivedList[position].question
            val drawableResourceId: Int=binding.root.resources.getIdentifier(
                receivedList[position].picPath,
                "drawable",
                binding.root.context.packageName
            )

            Glide.with(this@QuestionActivity)
                .load(drawableResourceId)
                .centerCrop()
                .apply(RequestOptions.bitmapTransform(RoundedCorners(60)))
                .into(pic)

            loadAnswer()
            rightArrow.setOnClickListener{
                if(progressBar.progress==10){
                    val intent= Intent(this@QuestionActivity, ScoreActivity::class.java)
                    intent.putExtra("Score", allScore)
                    startActivity(intent)
                    finish()
                    return@setOnClickListener
                }
                position++
                progressBar.progress=progressBar.progress+1
                questionNumberTxt.text = "Question " + progressBar.progress + "/10"
                questionTxt.text=receivedList[position].question

                val drawableResourceId: Int=binding.root.resources.getIdentifier(
                    receivedList[position].picPath,
                    "drawable", binding.root.context.packageName
                )

                Glide.with(this@QuestionActivity)
                    .load(drawableResourceId)
                    .centerCrop()
                    .apply(RequestOptions.bitmapTransform(RoundedCorners(60)))
                    .into(pic)

                loadAnswer()
            }
            leftArrow.setOnClickListener{
                if(progressBar.progress==1){
                    return@setOnClickListener
                }

                position--
                progressBar.progress=progressBar.progress-1
                questionNumberTxt.text = "Question " + progressBar.progress + "/10"
                questionTxt.text=receivedList[position].question

                val drawableResourceId: Int=binding.root.resources.getIdentifier(
                    receivedList[position].picPath,
                    "drawable", binding.root.context.packageName
                )

                Glide.with(this@QuestionActivity)
                    .load(drawableResourceId)
                    .centerCrop()
                    .apply(RequestOptions.bitmapTransform(RoundedCorners(60)))
                    .into(pic)

                loadAnswer()

            }
        }
    }

    private fun loadAnswer(){
        val users: MutableList<String> = mutableListOf()
        users.add(receivedList[position].answer_1.toString())
        users.add(receivedList[position].answer_2.toString())
        users.add(receivedList[position].answer_3.toString())
        users.add(receivedList[position].answer_4.toString())

        if (receivedList[position].clickedAnswer ! = null) users.add(receivedList[position].clickedAnswer.toString())

        val questionAdapter by lazy {
            QuestionAdapter(
                receivedList[position].correctAnswer.toString(), users, this
            )
        }
    }
    override fun amount(number: Int, clickedAnswer: String){
        allScore = number
        receivedList[position].clickedAnswer = clickedAnswer

    }
}